### DISCUSSION TITLE
npm stylelint: need help to run it

### DISCUSSION THREAD

**Discussion by math-GH** [FreshRSS/FreshRSS#4602] [General]
Hi,

I am very fresh with npm and want to run stylelint.

`npm run rtlcss` and `npm run eslint` works fine. But `npm run stylelint` does not work for me (on a Windows 10 machine):
![grafik](https://user-images.githubusercontent.com/1645099/189536679-6982beda-9beb-488d-a101-3a18637e7b6d.png)

(Stylelint 14.9.1 - given by the actual package-lock.json)

Could you help me please?

**Comment by math-GH** [comment]
After an `npm update` that changed stylelint to `14.11.0`:

![grafik](https://user-images.githubusercontent.com/1645099/189537244-b9bcefc6-4a3c-461d-ae10-195f14e33456.png)

**Comment by Alkarex** [comment]
On Windows, I would strongly suggest using WSL. Since ~2017, this has allowed running a native Linux development on Windows, much more easily and reliably.

* https://ubuntu.com/wsl
* https://docs.microsoft.com/windows/wsl
* https://code.visualstudio.com/docs/remote/wsl (nicely integrated in Visual Studio Code)

I sent you a short guide by e-mail some months ago. I can gladly help some more, e.g. by teleconference, if needed.

Otherwise, if you run Docker, we could produce a Docker image ready to run. 

I have not tried stylelint from a DOS-style terminal, and I am not even sure it is compatible. But if it is, the pattern `**/*.css` might have to be adapted to something that a Windows shell can understand.

**Reply by math-GH** [reply]
Thanks. WSL is on my todo list. Before, I started to learn a little bit to understand npm. It is part of my learning journey.

**Reply by Alkarex** [reply]
Ok :-) It is much easier to run npm on WSL than on DOS-style terminal

**Comment by math-GH** [comment]
OK, I was kicked [again ](https://github.com/FreshRSS/FreshRSS/issues/4268) with the CLI of Windows :/

see:
https://github.com/stylelint/stylelint/issues/3624

I need to change in `package.json` to
`"stylelint": "stylelint \"**/*.css\" && stylelint --custom-syntax postcss-scss \"**/*.scss\"",`

Without `'` it works too:
`    "stylelint": "stylelint **/*.css && stylelint --custom-syntax postcss-scss **/*.scss",`

**Comment by Frenzie** [comment]
Windows 10 machine is a bit vague. Is that cmd, weirdly ancient PowerShell that Windows ships by default for some reason or modern pwsh?

Anyway, the alternative for `**/*.css` is likely to be just `*.css` unless there's an alternative for **.

**Reply by math-GH** [reply]
> Windows 10 machine is a bit vague. Is that cmd, weirdly ancient PowerShell that Windows ships by default for some reason or modern pwsh?

The issue is on CMD and on PowerShell. It makes no difference here.

**Comment by math-GH** [comment]
My proof of concept for Windows:
see #4607
